### PR TITLE
[GP-3780] Clarify the boolean argument in ShesmuImputSource methods

### PR DESCRIPTION
**Pull request by avarsava**
JIRA Ticket: GP-3780

- [ ] Updates Changelog
- [X] Updates developer documentation

### CODE REVIEW

File: implementation.md [hunk 1]
```diff
@@ -597,9 +597,22 @@ exist. Suppose the type of the format is _T_. To add a source of data:
...
-is free to not make the effort to refresh it.
+The Shesmu server will call the `@ShesmuInputSource` in different situations,
+including at the start of olive execution, and when a simulator request
+is made. Different situations have different needs with regards to how 
+fresh the input data should be.
+The `boolean` argument, if included, is sent by the Shesmu server to 
+indicate to the plugin whether 
+the fetch should trigger a refresh. If true, the server has indicated that
+it needs fresh data. If false, the server is happy to 
+read stale data and the input source is free to not make the effort
```
Comment (djcooke) [inline implementation.md:608]: Might be clearer to say "older" instead of "stale" here given the other ways we use the term "stale"

**Review by djcooke — APPROVED** [review]

File: implementation.md [hunk 2]
```diff
@@ -597,9 +597,22 @@ exist. Suppose the type of the format is _T_. To add a source of data:
...
+fresh the input data should be.
+The `boolean` argument, if included, is sent by the Shesmu server to 
+indicate to the plugin whether 
+the fetch should trigger a refresh. If true, the server has indicated that
+it needs fresh data. If false, the server is happy to 
+read stale data and the input source is free to not make the effort 
+to refresh it. 
+
+Input source data is not cached by the server between calls to the 
+@ShesmuInputSource method. Caching data is the plugins responsibility.
```
Comment (djcooke) [inline implementation.md:612]: plugins -> plugin's or plugins'

File: implementation.md (hunk 1 above)
Comment (avarsava) [reply implementation.md]: Reworded, thanks

File: implementation.md (hunk 2 above)
Comment (avarsava) [reply implementation.md]: Fixed!

File: implementation.md [hunk 3]
```diff
@@ -597,9 +597,22 @@ exist. Suppose the type of the format is _T_. To add a source of data:
...
+fresh the input data should be.
+The `boolean` argument, if included, is sent by the Shesmu server to 
+indicate to the plugin whether 
+the fetch should trigger a refresh. If true, the server has indicated that
+it needs fresh data. If false, the server is happy to 
+receive data cached from a previous fetch and the input source is 
+free to not make the effort to refresh it. 
+
+Input source data is not cached by the server between calls to the 
+@ShesmuInputSource method. Caching data is the plugin's responsibility.
```
Comment (callunity) [inline implementation.md:612]: Do you want to include backticks around @ShesmuInputSource here like you did above?

**Review by callunity — APPROVED** [review]

**Review by grapigeau — APPROVED** [review]

**Pull request merged**